cache/memcached_client.go

```diff
@@ -196,13 +203,16 @@ func newMemcachedClient(
 		addressProvider: addressProvider,
 		stop:            make(chan struct{}, 1),
 		getMultiGate: gate.New(
-			prometheus.WrapRegistererWithPrefix("memcached_getmulti_", reg),
+			promregistry.TeeRegisterer{
+				prometheus.WrapRegistererWithPrefix(getMultiPrefix, reg),
+				prometheus.WrapRegistererWithPrefix(getMultiPrefix, newRegisterer),
+			},
 			config.MaxGetMultiConcurrency,
 		),
 	}
 
 	c.clientInfo = promauto.With(reg).NewGaugeFunc(prometheus.GaugeOpts{
```

Comment:
Here you should use `backwardCompatibleRegs` otherwise it's a breaking change.

Reply:
Actually, I am not adding `backend="memcached"` label intentionally to keep this metric in the old format because `client_info` labels are really client specific and removing memcached from metric name to label might make it confusing. Both redis and memcached will have different set of labels. In this line, `reg` (now renamed to `legacyReg` already has `memcached_` prefix, hence this line will produce the following as in the old format:

```
thanos_memcached_client_info{component="query-frontend",dns_provider_update_interval="30s",max_async_buffer_size="25000",max_async_concurrency="50",max_get_multi_batch_size="100",max_get_multi_concurrency="100",max_idle_connections="100",max_item_size="1048576",name="frontend-cache",timeout="500ms"} 1
versioned_client_memcached_client_info{dns_provider_update_interval="30s",max_async_buffer_size="25000",max_async_concurrency="50",max_get_multi_batch_size="100",max_get_multi_concurrency="100",max_idle_connections="100",max_item_size="1048576",name="admin-cache",timeout="200ms"} 1
```
For redis, I want to have the following format (in this commit still inconsistently produce `versioned_client_cache_redis_client_info`):
```
versioned_client_redis_client_info{backend="redis",connection_pool_size="100",dial_timeout="5s",max_async_buffer_size="25000",max_async_concurrency="50",max_get_multi_batch_size="100",max_get_multi_concurrency="100",max_item_size="1048576",name="admin-cache",read_timeout="3s",write_timeout="3s"} 1
```

Definitely we will maintain backward compatible change for memcached client_info, but do you think we still need to remove cache from metric name and move it to label?

Reply:
You're right. What you did is correct. Thanks!